### PR TITLE
refactor: move linkup from YAML to JSON

### PR DESCRIPTION
**Pull request by augustoccesar**
### Description

This PR moves from YAML to JSON.

Note: the tests are currently not passing on `main`, and consequently not passing here. Haven't had time to look into it.

### CODE REVIEW

File: e2e/src/main.rs [hunk 1]
```diff
@@ -49,44 +49,62 @@ impl Drop for Cleanup {
...
-  domains:
-    - domain: somedomain.com
-      default_service: frontend
-      routes:
-        - path: /api/v1/.*
-          service: backend
-    - domain: api.somedomain.com
-      default_service: backend
-      "#;
+    {
```
Comment (ostenbom) [inline e2e/src/main.rs:52]: This should still be yaml - the local configs -> yaml, things the server handles -> JSON

File: linkup-cli/src/local_config.rs [hunk 2]
```diff
@@ -48,9 +48,9 @@ impl Display for ServiceTarget {
 }
 
 #[derive(Deserialize)]
-pub struct YamlLocalConfig {
+pub struct JsonLocalConfig {
```
Comment (ostenbom) [inline linkup-cli/src/local_config.rs:51]: I think this should still be Yaml!

File: linkup/src/lib.rs [hunk 3]
```diff
@@ -229,30 +229,48 @@ mod tests {
...
-        location: http://localhost:8001/
-    domains:
-      - domain: example.com
-        default_service: frontend
-        routes:
-          - path: /api/v1/.*
-            service: backend
-      - domain: api.example.com
-        default_service: backend
+    {
```
Comment (ostenbom) [inline linkup/src/lib.rs:232]: This = yes!

File: e2e/src/main.rs (hunk 1 above)
Comment (augustoccesar) [reply e2e/src/main.rs:52]: Aha! Will take a look into it then!

**Pull request merged**